package-builder/build-manifest.nix

```diff
@@ -186,40 +186,58 @@ pkgs.runCommandNoCC name
         exit 1
       fi
 
+      # Take inventory of executables found in bin, sbin, and libexec.
+      declare -a bin sbin libexec;
+      shopt -s nullglob
+      for i in bin sbin libexec; do
+        for j in $out/$i/*; do
+          if [ -x "$j" ]; then
+            eval "$i+=($j)"
+          else
+            # Non-executable files are permitted in libexec but
+            # not in bin or sbin.
+            if [ "$i" != "libexec" ]; then
+              echo "⚠️  WARNING: $j is not executable." 1>&2
+            fi
+          fi
+        done
+      done
+
       # Check if there are binaries in $out/bin
-      mapfile -t binaries_in_bin < <(
-        fd --base-directory "$out" './bin/*' --type executable \
-        fd --base-directory "$out" './sbin/*' --type executable
-      )
-      if [ "''${#binaries_in_bin[@]}" -eq 0 ]; then
+      if [ "''${#bin[@]}" -eq 0 ]; then
```

Comment:
Yeah I thought of that and while we _could_ trigger when we found nothing at all to wrap, a package with files in `sbin` and/or `libexec` but not in `bin` would be so unusual as to probably be a mistake. I'm happy to carry on with that assumption for now and wait for the user feedback to educate us otherwise!

Reply:
> add a test for non executables in bin — not super blocking but it’s new behavior I think

Good point - previously the `nix build` would fail on account of the `assertExecutable` call on line 244 and now we're just sailing through with a warning instead.

Reply:
I updated the existing `build_no_bin_in_out_bin_sandbox_{,im}pure` tests in commit 19459622ead02792d23e24dba32a170fbd9332a9 to add a non-executable file in `bin` and confirm that the warning message appears. With that I'm inclined to rename those tests to `build_no_executable_files_in_out_bin_sandbox_{,im}pure` but otherwise I think we're good to go.

Reply:
Renamed in commit d73492beb11794bc7368e83f44ea39549e8f9c0c

Now adding comment as requested by @ysndr offline

Reply:
Discussed with @ysndr renaming the tests to be `build_verify_sane_out_sandbox_{off,pure}` to more fully indicate the general-purposeness of the test in validating the "sanity" of the build output.

We're now testing for:
* warning about non-files in `{bin,sbin}` (note we don't warn for `libexec`)
* warning about non-executables in `{bin,sbin}` (note we don't warn for `libexec`)
* warning about no executables found in `bin` only
* flagging executables found in directories other than `{bin,sbin,libexec}`
    - which includes flagging executable found in subdirectories of `{bin,sbin,libexec}`

Reply:
OK .. after more offline discussion consolidated tests into a single function, added comments, squashed commits and we're now ready to merge. Resolving conversation.